hazelcast/src/main/java/com/hazelcast/datastore/impl/ExternalDataStoreServiceImpl.java

```diff
@@ -64,4 +66,17 @@ public ExternalDataStoreFactory<?> getExternalDataStoreFactory(String name) {
         }
         return externalDataStoreFactory;
     }
+
+    public void shutDown(boolean terminate) {
+        if (terminate) {
+            return;
+        }
+
+        for (ExternalDataStoreFactory<?> dataStoreFactory : dataStoreFactories.values()) {
+            Object dataStore = dataStoreFactory.getDataStore();
+            if (dataStore instanceof Closeable) {
```

Comment:
If `shared` is false this creates a new instance of the data store, I suppose we don't want to do that on shutdown.